tests/tests/or_else.cpp

```diff
@@ -437,27 +415,38 @@ template <typename T> constexpr auto fn_int_lvalue = [](int &) -> T { throw 0; }
 template <typename T> constexpr auto fn_int_rvalue = [](int &&) -> T { throw 0; };
 } // namespace
 
-// clang-format off
 static_assert(invocable_or_else<decltype(fn_Error<std::expected<Value, Error>>), std::expected<Value, Error>>);
 static_assert(invocable_or_else<decltype(fn_generic<std::expected<int, int>>), std::expected<int, int>>);
-static_assert(invocable_or_else<decltype(fn_Error<std::expected<Value, Xerror>>), std::expected<Value, Error>>); // error type conversion
-static_assert(invocable_or_else<decltype(fn_Error<std::expected<Value, Error>>), std::expected<Value, Xerror>>); // error type conversion
-static_assert(invocable_or_else<decltype(fn_Error<std::expected<Value, int>>), std::expected<Value, Xerror>>); // error type conversion
-static_assert(not invocable_or_else<decltype(fn_Error<std::expected<Value, int>>), std::expected<Value, int>>); // wrong error_type
+static_assert(invocable_or_else<decltype(fn_Error<std::expected<Value, Xerror>>),
```

Comment:
Hm. Might roll these back. On one hand i dislike manual overrides, on the other hand this is way less readable :rage2: